code/modules/vehicles/mecha/mecha_defense.dm

```diff
@@ -382,7 +382,7 @@
 	for(var/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/gun in flat_equipment)
 		ammo_needed = 0
 
-		if(!istype(gun, /obj/item/mecha_parts/mecha_equipment/weapon/ballistic) && gun.ammo_type == A.ammo_type)
+		if(!istype(gun, /obj/item/mecha_parts/mecha_equipment/weapon/ballistic) || gun.ammo_type != A.ammo_type)
```

Comment:
when will this istype fail, the for() already checks the type?

Reply:
this check is as old as time (#44529) apparently and it is in fact redundant wow
https://github.com/tgstation/tgstation/commit/fe19e9f3ebc748120a9742bcc659c12f6df5f090